src/main/java/appeng/items/materials/ItemMaterial.java

```diff
@@ -311,7 +312,7 @@ else if( te instanceof IUpgradeableHost )
 
 				if( u != null )
 				{
-					final InventoryAdaptor ad = InventoryAdaptor.getAdaptor( upgrades, EnumFacing.UP );
+					final InventoryAdaptor ad = new AdaptorItemHandler( upgrades );
 					if( ad != null )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2589)